### ISSUE TITLE
IndexOutOfBounds on Queue.first() while size > 0

### ISSUE THREAD

**Issue by Robbiedobbie** [libgdx/libgdx#4300]
#### Issue details

An IndexOutOfBounds exception occurs when queue.first() is called. This seems to only happen when removing all/the last item in a queue through an iterator, and then adding a new item to the queue again.
#### Reproduction steps/code

```
import java.util.Iterator;

import org.junit.Test;

import com.badlogic.gdx.utils.Queue;

public class LibgdxQueueBug {

    @Test
    public void testQueue() {
        Queue<Integer> queue = new Queue<Integer>();

        //Simulate normal usage
        for(int i = 0; i < 100; i++) {
            queue.addLast(i);
            if(i > 50)
                queue.removeFirst();
        }

        Iterator<Integer> it = queue.iterator();
        while(it.hasNext()) {
            it.next();
            it.remove();
        }

        queue.addLast(1337);

        Integer i = queue.first();
    }
}
```
#### Version of LibGDX and/or relevant dependencies

Tested in LibGDX version 1.9.3, but looking at the git repo, the queue is not changed since.
#### Stacktrace

``` java
java.lang.ArrayIndexOutOfBoundsException: 64
    at com.badlogic.gdx.utils.Queue.first(Queue.java:256)
    at rails.LibgdxQueueBug.testQueue(LibgdxQueueBug.java:28)
    ...
```
#### Please select the affected platforms
- [x] Android
- [x] iOS (robovm)
- [x] iOS (MOE)
- [x] HTML/GWT
- [x] Windows
- [x] Linux
- [x] MacOS

**Comment by Darkyenus** [comment]
Thanks for great bug report! Fixed in #4302.

**Comment by badlogic** [comment]
Merged #4302
